src/components/PopupWithForm.js

```diff
@@ -7,9 +7,17 @@ export default class PopupWithFrom extends Popup {
     this._popupForm = this._popupElement.querySelector(".modal__form"); // selects modal form for specific instance of this obj
   }
 
+  setLoading(isLoading, submitSave) {
+    if (isLoading) {
+      this._popupElement.querySelector(".modal__save-button").textContent = "Saving...";
+    } else {
+      this._popupElement.querySelector(".modal__save-button").textContent = submitSave;
+    }
+  }
+
   close() {
-    this._popupForm.reset();
     super.close();
+    this._popupForm.reset();
     this._popupElement.removeEventListener("submit", this._handleFormSubmit);
```

Comment:
`    this._popupElement.removeEventListener("submit", this._handleFormSubmit);
`

You don't set `_handleFormSubmit` as the submit handler. You set `_submitForm` in line 42, so you need to remove `_submitForm`



```js

    this._popupElement.removeEventListener("submit", this._submitForm);

```